ledger/streaming/client/src/scp_validator.rs

```diff
@@ -206,13 +209,13 @@ impl<ID: GenericNodeId + Send + Clone> SCPValidationState<ID> {
                             }
                         }
                     }
-                    QuorumSetMember::InnerSet(qs) => {
+                    Some(QuorumSetMember::InnerSet(qs)) => {
                         // If internal slice reached quorum, record the block voted for
                         if let Some(block_id) = self.find_quorum(qs, index) {
                             ballot_map
                                 .entry(block_id)
-                                .and_modify(|vec| vec.push(member))
-                                .or_insert_with(|| vec![member]);
+                                .or_default()
+                                .push(member.unwrap());
```

Comment:
Similar to above, consider omitting the `unwrap()` in favor of:
```suggestion
                    Some(inner_set @ QuorumSetMember::InnerSet(qs)) => {
                        // If internal slice reached quorum, record the block voted for
                        if let Some(block_id) = self.find_quorum(qs, index) {
                            ballot_map
                                .entry(block_id)
                                .or_default()
                                .push(inner_set);
```